pkg/results/error.go

```diff
@@ -0,0 +1,83 @@
+package results
+
+import (
+	"fmt"
+)
+
+// Error holds a message and a child, allowing for an error
+// The common use-case here will be to wrap errors from callsites:
+//
+// if err := doSomething(data); err != nil {
+//     return results.ForReason(results.ReasonFoo).WithError(err).Errorf("could not do something for data: %v", data)
+// }
+type Error struct {
+	reason  Reason
+	message string
+	wrapped error
+}
+
+// Error makes an Error an error
+func (e Error) Error() string {
+	return e.message
```

Comment:
This can lead to non-nil errors that have an empty message. Please print the reason here and only append the `message` if its not an empty string

Reply:
Only if people are not using the builder chain correctly. That's not the point of this field.